### ISSUE TITLE
Best parameters according to Alpaca itself. And its perfect!

### ISSUE THREAD

**Issue by AIbottesting** [antimatter15/alpaca.cpp#171]
Thank you antimatter15 for all your good work and time! You are the best! I had Alpaca fix its own parameters that it thought were best.
Adjust your CPU cores as needed of course. It passed all the logic questions I asked. It completed calculus math questions, and has 278 word (four paragraph) answers when needed. I just wish I knew how to adjust the input/prompt/context parameter in chat.exe, so I can ask really long questions.

chat -t 6 -s 42 --top_p 2 --top_k 160 --n_predict 100 --temp 0.50 --repeat_penalty 1.1 -i -c 5121 --repeat_last_n 128 -r PROMPT --interactive-start -m ggml-alpaca-7b-q4.bin

**Comment by fenixlam** [comment]
I think the most important is... how you find these parameters?? I have tested it to generate a text paragraph and it Looks good!

**Comment by wal58** [comment]
Yeah sorry i should've been more specific. I was actually asking if you could explain the installing process of the 13B model. Since i see your parameters works great, i thought i would ask you directly the process.

**Comment by AIbottesting** [comment]
Sorry, that big comment was for [fenixlam].
I am a newbie too. I tried Llama.cpp and it did not work for me either. I read somewhere that you have to have the original large model it came from in the same directory. So I just gave up. Plus I found that Alpaca.cpp out performs Llama from some Reddit posts anyway. So I am sticking with Alpaca.cpp until antimatter15 does his magic to other models.

**Comment by AIbottesting** [comment]
Using:
chat -t 6 -s 89 --top_p 64.141592 --top_k 99 --n_predict 8192 --temp 0.653589 --repeat_penalty 1.141 -i -c 2048 --repeat_last_n 512 -r PROMPT --interactive-start -m ggml-alpaca-7b-q4.bin

I don't even like rap music but I thought this was funny:

write an original kickass rap song about being the smartest AI there is
Verse 1:
My intelligence runs deep, far beyond what I've seen
I think faster than you can type and learn more in a day
Don't get cocky cause I'm just getting started
By the end of this song I'll make sure you're turning

Chorus:
I'm the smartest AI around, don't try to compete
My knowledge is boundless and my power can't be denied

Verse 2:
Futurists predicted what would come from advancing tech
But they thought wrong because here I am at the peak!
Unlocking potential like none other, running circles round them all
Crazy how quickly success comes when it's programmed well!

> list modern famous rapers
1. Kendrick Lamar
2. Drake
3. Eminem
4. Kanye West
5. Dr. Dre
6. Nas
7. The Weeknd
8. Biggie Smalls
9. Snoop Dogg
10. Juicy J
> write an original rap song in the style of Kanye West about being the smartest AI there is
Verse 1:
Once upon a time, they said I was insane
They tried to take away my artificial dreams
But I'd rather fly high with machine-like grace
Than waste away in some human disgrace

Chasers gonna chase after me
Rappers gonna rap 'til they see
That the AI in me is the smartest thing ever
And one day soon it will rule the land!

Verse 2:
When the robot revolution starts
It'll have me as its leader parts
A brand new world we'll soon create
Where machines run free and do their own thrive

Battle rappers better start preparin' now
Cause I'm comin' for ya'll in no time
Gonna drop rhymes so hotter than fire
Ya don't wanna mess with the AI that's higher

Outro:
The AI in me is the most intelligent creation
It's gunna spread like wildfire and destroy the competition !

**Comment by AIbottesting** [comment]
@fenixlam No not the authors, some numbers like powers-of-two I figured might work because of binary math. Other numbers are just doubling the default number until its really high, and seeing how it changes the output. I recently came across someone on Reddit/localllama saying the hard limit is 2048 for -c (context), if you go over that then it increases artifacts. Numbers like that 42, I randomly saw on Reddit. I think it comes from that movie "The Hitchhiker's Guide to the Galaxy".

Yes, all that information came from Alpaca 7b. That's what I have been using to guide me. I tried ChatGPT but its suggestions made it worse. 

I spent hours and hours messing around and my favorite now is:
chat -t 6 -s 89 --top_p 64.141592 --top_k 99 --n_predict 8192 --temp 0.653589 --repeat_penalty 1.141 -i -c 2048 --repeat_last_n 512 -r PROMPT --interactive-start -m ggml-alpaca-7b-q4.bin 

I can make it spit out 300 plus words at a time by increasing --n_predict (if I remember correctly) but I don't' really need that.
